### ISSUE TITLE
[useToggle] Rewrite as a wrapper hook

### ISSUE THREAD

**Issue by kripod** [kripod/standard-hooks#36]
## Motivation

While the current implementation of `useToggle` handles a given state variable specially, [wrapper syntax](https://www.youtube.com/watch?v=nUzLlHFVXx0) would allow it to control custom states (e.g. the one returned by `useLocalStorage`).

## Basic example

```ts
function Example() {
  const [isPressed, togglePressed] = useToggle(useLocalStorage('isPressed', false));
}
```

**Comment by kripod** [comment]
Based on the proposed interface of [`useCounter`](https://github.com/kripod/standard-hooks/issues/37), an alternative API would be as follows:

```ts
function Example() {
  const [isPressed, setPressed, togglePressed] = useToggle(
    useLocalStorage('isPressed', false)
  );
}
```

**Comment by Jfelix61** [comment]
Hi, I would like to try this

**Comment by kripod** [comment]
@Jfelix61 Thanks for your contribution! If you need any kind of assistance, I'm glad to help.

**Comment by Jfelix61** [comment]
@kripod, just saw react-use repo and they had a hook called useBoolean which basically the same as useToggle. However, we can name our wrapper syntax useToggle as useBoolean and give useToggle its old interface, so the users can have options.

**Comment by kripod** [comment]
@Jfelix61 Personally, I don't think it's a good idea to have those two implemented with different functionality, as it may cause inevitable confusion among developers.

**Comment by Jfelix61** [comment]
> @Jfelix61 Personally, I don't think it's a good idea to have those two implemented with different functionality, as it may cause inevitable confusion among developers.

Yea you are right, then I'll leave it as it is.

**Comment by kripod** [comment]
This has been resolved by #78.

@Jfelix61 If you ever decide to contribute again, please refer the corresponding issue (e.g. this one) in your pull request [using keywords](https://help.github.com/en/articles/closing-issues-using-keywords) if possible. Thank you for your efforts!
